CarpetX/src/schedule.cxx

```diff
@@ -2005,6 +2026,15 @@ int CallFunction(void *function, cFunctionData *restrict attribute,
 
 bool sync_active = false; // Catch recursive calls
 
+struct mark_sync_active {
+    mark_sync_active() {
+        sync_active = true;
```

Comment:
It would be good to add `assert` statements here that catch the case that sync is already active when it's activated, or that it's inactive when it's deactivated.